server/src/com/cloud/api/query/dao/HostJoinDaoImpl.java

```diff
@@ -154,8 +154,10 @@ public HostResponse newHostResponse(HostJoinVO host, EnumSet<HostDetails> detail
                 Long mem = host.getMemReservedCapacity() + host.getMemUsedCapacity();
                 Long cpu = host.getCpuReservedCapacity() + host.getCpuUsedCapacity();
 
+                hostResponse.setMemoryTotal(mem);
+                Float totalMemorywithOverprovisioning=new Float((host.getTotalMemory()*ApiDBUtils.getMemOverprovisioningFactor(host.getClusterId())));
+                hostResponse.setMemWithOverprovisioning(totalMemorywithOverprovisioning.toString());
```

Comment:
Hey guys,

Looks like there is a flaw with the logic. We have a cluster of XenServer hosts with 512 GB of RAM and the UI shows ~170 GB of RAM. This is not even including the overprovisioning factor. Can you please check the logic around here

here `getMemReservedCapacity` is for most cases 0 which leads to `mem = host.getMemUsedCapacity()` and later  `hostResponse.setMemoryTotal(mem)` which is wrong. Can you please explain the logic here? @bvbharatk @karuturi ?

```

Reply:
@syed I agree. I think this should just be `hostResponse.setMemoryTotal(host.getTotalMemory());`